django_project/frontend/views/users.py

```diff
@@ -183,36 +184,41 @@ def invite_post(self, request):
             )
             org_id = get_current_organisation_id(request.user)
             if not is_new_invitation:
-                if role == 'manager':
-                    create_invite = OrganisationInvites(
-                        email=email,
-                        organisation_id=org_id,
-                        user_role=user_role,
-                        assigned_as=MANAGER
-                    )
-                else:
-                    create_invite = OrganisationInvites(
-                        email=email,
-                        organisation_id=org_id,
-                        user_role=user_role,
-                        assigned_as=MEMBER
-                    )
-
                 organisation = Organisation.objects.get(id=org_id)
 
                 # assert if user is registered on platform
-                registered = self.is_user_registerd(email)
+                registered, user = self.is_user_registered(email)
+                invitation_uuid = uuid.uuid4().hex
                 if registered:
-                    encoded_email = quote(email, safe='')
                     return_url = (
                         Site.objects.get_current().domain +
-                        f'/adduser/{encoded_email}/{organisation.name}/'
+                        f'/adduser/{invitation_uuid}/'
                     )
                 else:
+                    encoded_email = quote(email, safe='')
                     return_url = (
                         Site.objects.get_current().domain +
-                        '/accounts/signup/?email=' + quote(email) +
-                        '&organisation=' + quote(organisation.name)
+                        '/accounts/signup/?email=' + encoded_email +
+                        '&uuid=' + quote(invitation_uuid)
+                    )
+
+                if assign_as == 'manager':
+                    create_invite = OrganisationInvites(
+                        email=email,
+                        organisation_id=org_id,
+                        user_role=user_role,
+                        assigned_as=MANAGER,
```

Comment:
you can do if else in the assigned_as if there is not much diff between two branches,

Reply:
@danangmassandy Updated